packages/babel-helper-compilation-targets/src/index.ts

```diff
@@ -254,17 +254,20 @@ export default function getTargets(
 
     if (esmodules === "intersect") {
       for (const browser of Object.keys(queryBrowsers) as Target[]) {
-        const version = queryBrowsers[browser];
-        const esmSupportVersion =
-          // @ts-expect-error ie is not in ESM_SUPPORT
-          ESM_SUPPORT[browser];
-
-        if (esmSupportVersion) {
-          queryBrowsers[browser] = getHighestUnreleased(
-            version,
-            semverify(esmSupportVersion),
-            browser,
-          );
+        if (browser !== "deno" && browser !== "ie") {
```

Comment:
```suggestion
        // Every version of Deno supports ESM,
        // and no version of IE supports ESM.
        if (browser !== "deno" && browser !== "ie") {
```

Reply:
Yes. And `deno` is excluded because that would result to a query that browserslist does not accept, so `deno` is not in `native-modules.json`. The `@ts-ignore` here is fixed so if we add a new target but forget to update `native-modules.json`, TS will catch this issue.

In the long term we may consider provide the es modules support matrix with compat-table engines, to be consistent with other features. The esmodule merge can be performed on the compat-table engines so we don't have to deal with differences between browserslist and compat-table.